sdk/digitaltwins/Azure.DigitalTwins.Core/tests/QueryTests.cs

```diff
@@ -61,12 +60,118 @@ await foreach (string response in asyncPageableResponse)
+                    {
+                        await client.DeleteDigitalTwinAsync(roomTwinId).ConfigureAwait(false);
+                    }
                 }
                 catch (Exception ex)
                 {
                     Assert.Fail($"Test clean up failed: {ex.Message}");
                 }
             }
         }
+
+        // TODO:azabbasi: Once you enable this test by removing the Ignore attribute, make sure to record and update session records.
+        [Test]
+        [Ignore("Waiting for Azure.Core to fix the issue with AsPages helper method")]
+        public async Task Query_MultiplePages_Resuming_Success()
+        {
+            DigitalTwinsClient client = GetClient();
+
+            string floorModelId = await GetUniqueModelIdAsync(client, TestAssetDefaults.FloorModelIdPrefix).ConfigureAwait(false);
+            string roomModelId = await GetUniqueModelIdAsync(client, TestAssetDefaults.RoomModelIdPrefix).ConfigureAwait(false);
+
+            string roomTwinId = await GetUniqueTwinIdAsync(client, TestAssetDefaults.RoomTwinIdPrefix).ConfigureAwait(false);
+
+            int numberOfTwinsToCreate = 250;
+            IEnumerable<string> newTwinIds = getMultipleTwinIds(roomTwinId, numberOfTwinsToCreate);
+
+            try
+            {
+                //arrange
+
+                // Create room model
+                string roomModel = TestAssetsHelper.GetRoomModelPayload(roomModelId, floorModelId);
+                await client.CreateModelsAsync(new List<string> { roomModel }).ConfigureAwait(false);
+
+                // Create multiple room twins, with property "IsOccupied": true
+                // We do this to make sure the service is going to return multiple pages in the response.
+                List<Task> tasks = new List<Task>();
+
+                foreach (var twinName in newTwinIds)
+                {
+                    string roomTwin = TestAssetsHelper.GetRoomTwinPayload(roomModelId);
+                    tasks.Add(client.CreateDigitalTwinAsync(twinName, roomTwin));
+                }
+
+                Task.WaitAll(tasks.ToArray());
+
+                string queryString = "SELECT * FROM digitaltwins where IsOccupied = true";
+
+                // act
+                IAsyncEnumerable<Page<string>> asyncPageableResponse = client.QueryAsync(queryString).AsPages();
+                string continuationToken = null;
+
+                // Get the first page and store the continuation token.
+                await foreach (Page<string> page in asyncPageableResponse)
+                {
+                    Console.WriteLine(page.ContinuationToken);
+                    continuationToken = page.ContinuationToken;
+                    break;
+                }
+
+                // resume getting the query response by providing the continuation token.
+                IAsyncEnumerable<Page<string>> nextPages = client.QueryAsync(queryString).AsPages(continuationToken);
```

Comment:
The expected behavior for AsPages(continuationToken) is to resume the paging operation from that specific continuation token
Currently it always goes back to the first page.